### PR TITLE
docs: ui capabilities button PTL-1583

### PR DESCRIPTION
**Pull request by Gareth-Spencer-Genesis**
ATENTION! YOU SHOULD NOW BRANCH FROM PREPROD WHEN YOU UPDATE 

  - Please check the [internal contributions guide](https://www.notion.so/genesisglobal/Contributing-new-documentation-75953fb245f246ff872789035451a0c4) for details
  
__________

Thank you for contributing to the documentation.

Do the changes you have made apply to both Current and Previous versions?
No

Have you done a trial build to check all new or changed links?
Yes

Is there anything else you would like us to know?
No

**This week's exciting advice from the style guide**

- Write your headings in sentence case:

  - A good example
    This is a correct heading.
  - A Bad Example
    This is not a correct heading.

### CODE REVIEW

File: docs/001_develop/03_client-capabilities/008_interaction/checkbox.mdx
Comment (wjhendry) [inline docs/001_develop/03_client-capabilities/008_interaction/checkbox.mdx:1]: what does "visual opacity/visually opaque" mean? Greyed out? Invisible? Something else?

File: docs/001_develop/03_client-capabilities/008_interaction/button.mdx [hunk 1]
```diff
@@ -0,0 +1,169 @@
...
+|type|`string`|Defines the mode of the button.|
+|value|`string`|Defines the value associated with name.|
+
+The type attribute can be:
+* button (default) - performs an action when clicked
+* reset - can influence a group of components associated with the same form.
+
+### Examples (closing tag omitted):
+
+Attributes:
```
Comment (patrickoneill-genesis) [inline docs/001_develop/03_client-capabilities/008_interaction/button.mdx:134]: Properties and attributes should be in this format 

See
https://github.com/genesiscommunitysuccess/docs/pull/1853/files#diff-b69edbdc88e90870eeee016c48b131788c8f74fcda501b2fae84807575e101abR329

File: examples/ui/client-capabilities/interaction/button.js [hunk 2]
```diff
@@ -0,0 +1,14 @@
+import { CodeLabel, CodeSection } from '../../documentationBase';
+import { registerComponents } from '../../rapidImports';
+
+registerComponents();
+
+export default function ButtonDemo({ children, color }) {
+	return (
+		<CodeSection>
+			<CodeLabel>Button:</CodeLabel>
+			<rapid-button>
```
Comment (patrickoneill-genesis) [inline examples/ui/client-capabilities/interaction/button.js:10]: can we flesh out this example covering all the different attributes?

Comment (patrickoneill-genesis) [reply examples/ui/client-capabilities/interaction/button.js:10]: and events

File: docs/001_develop/03_client-capabilities/008_interaction/button.mdx [hunk 3]
```diff
@@ -0,0 +1,162 @@
...
+    ```
+    Usage
+    ```typescript
+    export function MyComponent() {
+      const handleButtonClicked = (event) => {
+        console.log('Button clicked')
+      };
+
+      return (
+        <rapid-button onClick={handleButtonClicked}></rapid-button>
```
Comment (patrickoneill-genesis) [inline docs/001_develop/03_client-capabilities/008_interaction/button.mdx:77]: This works in react - can we change the handler to `alert` like the other examples. 

Add some button text.

Can we add some examples of slots - start/end, and one with appearance value set

File: docs/001_develop/03_client-capabilities/008_interaction/button.mdx [hunk 4]
```diff
@@ -0,0 +1,162 @@
...
+        <rapid-button
+          (click)="onButtonClicked($event)"
+        ></rapid-checkbox>
+      `,
+      styleUrls: ['./my.component.css'],
+      standalone: true,
+      schemas: [CUSTOM_ELEMENTS_SCHEMA],
+      imports: [FormsModule],
+    })
+    export class AppComponent {
```
Comment (patrickoneill-genesis) [inline docs/001_develop/03_client-capabilities/008_interaction/button.mdx:105]: Tested and works in angular. Same as above - can we add the slots, appearance examples

File: docs/001_develop/03_client-capabilities/008_interaction/button.mdx [hunk 5]
```diff
@@ -0,0 +1,234 @@
...
+```typescript
+<rapid-button autofocus>
+```
+
+</td>
+    </tr>
+    <tr>
+      <td>appearance</td>
+      <td><code>string</code></td>
+      <td>Configures the component so the user cannot alter the state via the web browser.</td>
```
Comment (matteematt) [inline docs/001_develop/03_client-capabilities/008_interaction/button.mdx:135]: is this the correct description for appearance?

File: docs/001_develop/03_client-capabilities/008_interaction/checkbox.mdx
Comment (matteematt) [inline docs/001_develop/03_client-capabilities/008_interaction/checkbox.mdx:1]: Can you drop the changes in this file please? I am doing a PR now which is updating checkbox to align with our format currently

**Review by matteematt — COMMENTED** [review]
Good, just two small bits

**Review by matteematt — APPROVED** [review]

**Pull request merged**